### ISSUE TITLE
nonPublicPartOfAPI not accepting reportUnchanged=false through maven plugin

### ISSUE THREAD

**Issue by mrpotes** [revapi/revapi#130]
I'm looking to transition to revapi from clirr, but I have an API that triggers the `nonPublicPartOfAPI` check. I checked the documentation and updated my maven configuration as follows:

```
<analysisConfiguration>
  <revapi.java>
    <checks>
      <nonPublicPartOfAPI>
        <reportUnchanged>false</reportUnchanged>
      </nonPublicPartOfAPI>
    </checks>
  </revapi.java>
</analysisConfiguration>
```

However, even with this configuration the check still triggers a build failure.

On debugging the `NonPublicClassPartOfAPI` class, I found that the `initialize` method is called twice - once from `JavaApiAnalyzer#initialize`, and then again from the `JavaElementDifferenceAnalyzer` constructor.

I noticed from the debugger that the configuration in the `AnalysisContext` in the first call has a value for `reportUnchanged` defined, and the value of `NonPublicClassPartOfAPI#reportUnchanged` is correctly set to `false`, but in the second call, the value is not defined, but there is a definition in the underlying JSON object for `checks.nonPublicPartOfAPI.reportUnchanged`.

Looking at the two different call points, the `JavaApiAnalyzer#initialize` method does the following:

```
            if (c.getExtensionId() != null) {
                ModelNode checkConfig = analysisContext.getConfiguration().get("checks", c.getExtensionId());
                AnalysisContext checkCtx = analysisContext.copyWithConfiguration(checkConfig);
                c.initialize(checkCtx);
            }
```

However, this sub-context for the extension is not obtained in the `JavaElementDifferenceAnalyzer` - it seems to me it should be doing the same thing?

Thanks!

**Comment by metlos** [comment]
Thanks for reporting this! This was a really stupid error where the implementation did a completely different thing to what was documented and even tested for that wrong behavior. I guess I should sleep more :wink:

**Comment by metlos** [comment]
Fixed in `revapi-java-0.17.0` (usable only with `revapi-maven-plugin-0.10.2` and onwards).
